### PR TITLE
Rp/improve deleted odata feed error message

### PR DESCRIPTION
**Pull request by RynhardPietersen**
## Product Description

When users attempt to access deleted odata feeds, they are presented with a 500 error. This was a bug as the code was trying to work with a object that didn't exist. I fixed that bug and now the users will be presented with a 404 that will inform them that the url they are trying to use does not exist.

## Technical Summary

https://dimagi-dev.atlassian.net/browse/SS-207

## Safety Assurance

- [x] Risk label is set correctly
- [x] All migrations are backwards compatible and won't block deploy
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change
- [x] If QA is part of the safety story, the "Awaiting QA" label is used
- [x] I have confidence that this PR will not introduce a regression for the reasons below
The changes have been tested locally and on staging

### QA Plan

I tested this locally and on staging

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### CODE REVIEW

File: corehq/apps/api/odata/serializers.py [hunk 1]
```diff
@@ -29,39 +30,43 @@ def to_json(self, data, options=None):
...
+            config = self.get_config(config_id)
+            data['value'] = self.serialize_documents_using_config(
+                data.pop('objects'),
+                config,
+                table_id
+            )
+            return json.dumps(data, cls=DjangoJSONEncoder, sort_keys=True)
+
+        except KeyError as e:
+            raise Http404(e)
```
Comment (AmitPhulera) [inline corehq/apps/api/odata/serializers.py:69]: Instead of passing an error object, do you think it would be a good idea to pass a custom error message?

Comment (biyeun) [reply corehq/apps/api/odata/serializers.py:69]: Also, in order to raise a better custom error message, as @AmitPhulera, maybe this part of the OData spec helps: https://docs.oasis-open.org/odata/odata-json-format/v4.0/os/odata-json-format-v4.0-os.html#_Toc372793091

Can you return a response formatted like what's described in that doc and see how PowerBI deals with it?

Comment (RynhardPietersen) [reply corehq/apps/api/odata/serializers.py:69]: Yeah `data['objects']` doesn't exist and raised a 500.

**Review by biyeun — APPROVED** [review]
looks great!

**Pull request merged**